### PR TITLE
TCM-116 | Build dummy profile screen

### PR DESCRIPTION
**Pull request by halilyesilyurt**
### Summary

- [x] Completed dummy profile screen.
- [x] Added profile navigation to home.
- [x] Added settings navigation to profile.

### Issue

[TCM-116](https://hyfnla.atlassian.net/browse/TCM-116)

### Video/Screenshot

| Live Section | Virtual Section|
|--------|------|
|profile_live|profile_virtual|

DEMO

https://user-images.githubusercontent.com/30659854/119556875-2f42e280-bda8-11eb-8e65-c45170c33212.mp4

### CODE REVIEW

File: app/src/main/res/layout/fragment_profile.xml [hunk 1]
```diff
@@ -0,0 +1,141 @@
+<?xml version="1.0" encoding="utf-8"?>
+<com.trx.consumer.common.CommonView xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    app:backgroundColor="@color/white">
+
+    <com.trx.consumer.common.CommonView
```
Comment (onodude) [inline app/src/main/res/layout/fragment_profile.xml:9]: please use the `viewNavBar` and `btnBack` specs from `Subscriptions` screen, please mimic the `btnBack` specs for `btnSettings` in terms of layout

**Review by onodude — CHANGES_REQUESTED** [review]
looking good overall @halilyesilyurt left a minor comment about the top layout, also could you please revert irrelevant changes (some icon files got mixed up)

File: app/src/main/java/com/trx/consumer/screens/liveworkouttable/LiveWorkoutTableAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import android.view.ViewGroup
+import com.trx.consumer.R
+import com.trx.consumer.common.CommonRecyclerViewAdapter
+import com.trx.consumer.common.CommonViewHolder
+import com.trx.consumer.managers.LogManager
+import com.trx.consumer.models.common.WorkoutModel
+import com.trx.consumer.views.EmptyViewHolder
+import kotlinx.coroutines.CoroutineScope
+
+class LiveWorkoutTableAdapter(
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/screens/liveworkouttable/LiveWorkoutTableAdapter.kt:13]: Let's create a proprietary adapter for the `Profile` screen called `ProfileAdapter` that will handle displaying both `Live` and `Virtual` cells. Can check out [SearchResultAdapter](https://github.com/hyfn/xpass-android/blob/develop/app/src/main/java/com/xponential/xpass/screens/search/SearchResultAdapter.kt) from Xpass for reference

File: app/src/main/java/com/trx/consumer/screens/liveworkouttable/LiveWorkoutTableListener.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.trx.consumer.screens.liveworkouttable
+
+import com.trx.consumer.models.common.WorkoutModel
+
+interface LiveWorkoutTableListener {
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/screens/liveworkouttable/LiveWorkoutTableListener.kt:5]: We can use the existing `LiveWorkoutListener` and `VirtualWorkoutListener`

**Review by onodude — CHANGES_REQUESTED** [review]
looks like some listeners and adapters are duplicate of existing ones. we do not need new listeners and adapters for table ones

File: app/src/main/java/com/trx/consumer/screens/virtualworkouttable/VirtualWorkoutTableViewHolder.kt [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import com.trx.consumer.common.CommonButton
+import com.trx.consumer.common.CommonImageView
+import com.trx.consumer.common.CommonLabel
+import com.trx.consumer.common.CommonView
+import com.trx.consumer.common.CommonViewHolder
+import com.trx.consumer.extensions.load
+import com.trx.consumer.extensions.px
+import com.trx.consumer.models.common.VirtualWorkoutModel
+
+class VirtualWorkoutTableViewHolder(view: View) : CommonViewHolder(view) {
```
Comment (VivekV95) [inline app/src/main/java/com/trx/consumer/screens/virtualworkouttable/VirtualWorkoutTableViewHolder.kt:14]: Don't think we need the extra `ViewHolders` either, can use the existing `LiveWorkoutViewHolder` and `VirtualWorkoutViewHolder`

**Review by VivekV95 — CHANGES_REQUESTED** [review]
Some changes in approach

File: app/src/main/res/layout/fragment_profile.xml (hunk 1 above)
Comment (halilyesilyurt) [reply app/src/main/res/layout/fragment_profile.xml:9]: Completed.

File: app/src/main/java/com/trx/consumer/screens/liveworkouttable/LiveWorkoutTableAdapter.kt (hunk 2 above)
Comment (halilyesilyurt) [reply app/src/main/java/com/trx/consumer/screens/liveworkouttable/LiveWorkoutTableAdapter.kt]: Completed. Thanks for your advise :)

File: app/src/main/java/com/trx/consumer/screens/liveworkouttable/LiveWorkoutTableListener.kt (hunk 3 above)
Comment (halilyesilyurt) [reply app/src/main/java/com/trx/consumer/screens/liveworkouttable/LiveWorkoutTableListener.kt]: Completed.

**Comment by halilyesilyurt** [comment]
updated pr images and demo.

File: app/src/main/res/layout/fragment_profile.xml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/btnSessions"
+        app:textAllCaps="true"
+        tools:ignore="SmallSp"
+        tools:text="No Live Workouts found" />
+
+    <com.trx.consumer.common.CommonHudView
+        android:id="@+id/hudView"
+        style="@style/MatchParentEdgesViewStyle"
+        android:layout_width="wrap_content"
```
Comment (VivekV95) [inline app/src/main/res/layout/fragment_profile.xml:119]: hud should match parent for both width and height

**Review by VivekV95 — CHANGES_REQUESTED** [review]
Looking great, one more change from me, and these icons still need to be removed:
Screen Shot 2021-05-25 at 4 51 42 PM

File: app/src/main/res/layout/fragment_profile.xml (hunk 5 above)
Comment (halilyesilyurt) [reply app/src/main/res/layout/fragment_profile.xml]: Completed.

**Comment by halilyesilyurt** [comment]
> Looking great, one more change from me, and these icons still need to be removed:
> Screen Shot 2021-05-25 at 4 51 42 PM

Completed.

**Review by VivekV95 — APPROVED** [review]

**Review by onodude — APPROVED** [review]
thank you!

**Pull request merged**